api/request_source/request_source_plugin.proto

```diff
@@ -0,0 +1,28 @@
+syntax = "proto3";
+
+package nighthawk.request_source;
+
+import "google/protobuf/wrappers.proto";
+import "validate/validate.proto";
+
+// Configuration for DummyPluginRequestSource (plugin name: "nighthawk.dummy-request-source-plugin")
+// that does nothing
+message DummyPluginRequestSourceConfig {
+  // Dummy input value.
+  google.protobuf.DoubleValue dummy_value = 1;
+}
+
+// Configuration for RPCPluginRequestSource (plugin name: "nighthawk.rpc-request-source-plugin")
+// that uses rpc
+message RPCPluginRequestSourceConfig {
+  string uri = 1;
+}
+
+// Configuration for FileBasedPluginRequestSource (plugin name:
+// "nighthawk.file-based-request-source-plugin") that uses rpc
+message FileBasedPluginRequestSourceConfig {
+  string uri = 1;
```

Comment:
Silly perhaps, but at the first glance I thought this `uri` field might be used to point out a to-be-downloaded file for replay.  Maybe a rename and/or a helpful comment is good here? For example "request_header_base_uri" or some such. 

I'm not sure if it's intentional, but it is also worth explicitly calling out that with the way this is now implemented, the result of applying `uri` to a request header can be overridden by setting related request headers (":path, :authority") on a per-request basis in "records" from the file contents.

That leads to the thought: if this behavior is intended/desired, should we allow this uri to be empty? If the file is expected to fully supply this for every request, that would make sense. Maybe, in that case, this field should be optional (via `google.protobuf.StringValue`?), so we can make a distinction between empty and not set. 

Alternatively, it is also possible to make a change to ignore or error out when ambiguity arises based on `uri` and file-based content. But either way, I think we need to define / document the desired behaviour. 

Or, possibly even generalize this accross plugins, and enforce consistent behavior.

Reply:
Oh -- afterthought here; I think the gRPC plugin also behaves like this; so we're consistent, but we're not enforcing that consistency. This could perhaps also be facilitated by a base implementation. I don't want to bloat this PR though.